### PR TITLE
[WIP] Player locale based on user's geolocated IP address

### PR DESCRIPTION
**Pull request by ibuckshot5**
This PR allows custom Player Locale based on the user's geolocated IP.

**This is very WIP and feel free to tweak any part of this.**

## Description

I added a new method to pogom/utils, `get_player_locale`. This returns locale for the user's IP, as stated before.

## Motivation and Context

Having a set locale can flag accounts?

## How Has This Been Tested?

On my personal installation

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by SiteEffect** [comment]
Much better than the geopy try of #2057. The problem is that people using proxies or running on a server do not want to use the IP as reference but the search location.

File: pogom/account.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ class TooManyLoginAttempts(Exception):
     pass
 
 
+locale = get_player_locale()
```
Comment (sebastienvercammen) [inline pogom/account.py]: Don't make this a global. Memoize `get_player_locale` instead.

File: pogom/utils.py [hunk 2]
```diff
@@ -720,6 +720,19 @@ def get_args():
...
+    locale = {'country': 'US',
+                       'language': 'en',
+                       'timezone': 'America/Denver'}
+    try:
+        res = requests.get(url='http://freegeoip.net/json')
+        dict = json.loads(res.text)
+        locale['country'] = dict['country_code']
+        locale['timezone'] = dict['time_zone']
+    except Exception:
+        pass
```
Comment (sebastienvercammen) [inline pogom/utils.py]: Don't `pass`: at least log the exception, e.g.:
```
except Exception as e:
    log.exception('Exception while getting player locale: %s.', e)
```
before returning the default value.

File: pogom/utils.py [hunk 3]
```diff
@@ -720,6 +720,19 @@ def get_args():
 
     return args
 
+def get_player_locale():
+    locale = {'country': 'US',
+                       'language': 'en',
+                       'timezone': 'America/Denver'}
+    try:
+        res = requests.get(url='http://freegeoip.net/json')
```
Comment (sebastienvercammen) [inline pogom/utils.py]: Your request **needs** a timeout, otherwise it can hang indefinitely.

**Comment by friscoMad** [comment]
I do think that geolocation from map coords seem a better solution, it is more complex due to gmaps api but maybe we can use another api to do the same, using the server ip it is not better than using just english.

**Comment by SiteEffect** [comment]
@sebastienvercammen Can be closed due to merged #2057

**Comment by sebastienvercammen** [comment]
Closed because we'll be using #2057 + #2106.

**Pull request closed without merging**